### DISCUSSION TITLE
Commit history not showing commits before move of file - timeline

### DISCUSSION THREAD

**Discussion by angelaschmid** [gitkraken/vscode-gitlens#2960] [Q&A]
After a file rename the commit history can be viewed with `git log --follow <file>`.

Although I set `gitlens.advanced.fileHistoryFollowsRenames = true ` the timeline still only shows commits starting from the rename.

(the older commits are visible as blames in the file)

Is this setting
- meant to show the commit history after a file rename and does not work as expected
- or is it used to view something elsewhere (where/how)

**Comment by eamodio** [comment]
Which view are you looking at? You mentioned  "timeline" so you do you mean the Timeline view in VS Code? If so, that is a VS Code view not a GitLens view. The GitLens version of that Timeline view is the File History view, and the setting you reference does what you said for that view.

**Reply by angelaschmid** [reply]
Thanks for referring to use the File History (GitLens) view. Unfortunately this view also only shows the commits since the rename.
The Line History (GitLens) shows the blames correct of any changes before the file rename.

GitLens version 13.5.0
